### ISSUE TITLE
Make connection with DomainSets.jl

### ISSUE THREAD

**Issue by dlfivefifty** [complexvariables/ComplexRegions.jl#1]
This package is very similar in spirit to IntervalSets.jl and DomainSets.jl. It would be good to try to combine the two as much as possible, and reference these in the paper.

One key thing: DomainSets.jl views domains/sets as an "interface" that support the `in` command. It would be good to override `in` as appropriate, e.g., the following should work:
```julia
julia> c
Circle{Complex{Float64}} in the complex plane:
   centered at (1.0 + 0.0im) with radius 1.0, negatively oriented

julia> 2.0 in c
ERROR: MethodError: no method matching iterate(::Circle{Complex{Float64}})
Closest candidates are:
  iterate(::Core.SimpleVector) at essentials.jl:604
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:604
  iterate(::ExponentialBackOff) at error.jl:214
  ...
Stacktrace:
 [1] in(::Float64, ::Circle{Complex{Float64}}) at ./operators.jl:1055
 [2] top-level scope at REPL[11]:1
```

**Comment by tobydriscoll** [comment]
I will add a reference, as I was previously unaware of these two packages. I was caught up in trying to figure out the extent of domain support in `ApproxFun`, which seems related but more limited...?

The code uses `in` for membership in regions, but it's not supported for curves and paths. I'm not sure if you mean set membership or geometric membership. The former is ill-posed in floating point, and the latter changes the meaning of set membership attached to `in`.

```julia
 julia> 0 in unitdisk
true

julia> isinside(0,Circle(0,1))
true
```

Arguably I'm being inconsistent since I use "soft" tolerances for a lot of things, including region membership, but somehow the question of detecting whether a point is on a curve feels egregiously bad to me.

**Comment by tobydriscoll** [comment]
Addressed the reference issue in 94a7c29.
